### PR TITLE
tealdeer: add module

### PR DESCRIPTION
**Pull request by marsam**
### Description

Add https://github.com/dbrgn/tealdeer
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/tealdeer.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+in {
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.tealdeer = {
+    enable = mkEnableOption "Tealdeer";
+
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
```
Comment (sumnerevans) [inline modules/programs/tealdeer.nix:22]:
```suggestion
      default = { };
      defaultText = literalExpression "{ }";
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
LGTM otherwise

**Review by sumnerevans — APPROVED** [review]

**Comment by sumnerevans** [comment]
Oh, I think this needed an entry in the news file. @rycee are we still doing those? I haven't seen any lately (possibly because I keep forgetting to request that people add them :eyes:)

File: modules/programs/tealdeer.nix (hunk 1 above)
Comment (rycee) [reply modules/programs/tealdeer.nix:22]: Is there any particular need for the `defaultText` entry? I think the manual output should be decent even with just `default = {}`, although I haven't checked recently 🙂

**Comment by rycee** [comment]
@sumnerevans I typically still add the news entry when merging the PR. I added one for tealdeer a little while ago.

Dunno if those news entries are particularly useful, maybe not?

**Pull request merged**